### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by alindima**

### CODE REVIEW

File: text/0047-random-assignment-of-availability-chunks.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+## Summary
+
+Propose a way of randomly permuting the availability chunk indices assigned to validators for a given core and relay chain block,
+in the context of [recovering available data from systematic chunks](https://github.com/paritytech/polkadot-sdk/issues/598).
+
+## Motivation
+
+The relay chain node must have a deterministic way of evenly distributing the first ~(N_VALIDATORS / 3) systematic availability chunks to different validators,
+based on the session, relay chain block number and core.
+This is needed in order to optimise network load distribution as evenly as possible during availability recovery.
```
Comment (sandreim) [inline text/0047-random-assignment-of-availability-chunks.md]: I would rephrase this to something like: `The main purpose is fair distribution of bandwidth usage for availability recovery in general and in particular for systematic chunk holders.`

File: text/0047-random-assignment-of-availability-chunks.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+| **Authors**     | Alin Dima                                                                                   |
+
+## Summary
+
+Propose a way of randomly permuting the availability chunk indices assigned to validators for a given core and relay chain block,
+in the context of [recovering available data from systematic chunks](https://github.com/paritytech/polkadot-sdk/issues/598).
+
+## Motivation
+
+The relay chain node must have a deterministic way of evenly distributing the first ~(N_VALIDATORS / 3) systematic availability chunks to different validators,
```
Comment (sandreim) [inline text/0047-random-assignment-of-availability-chunks.md]: why ~ ?

File: text/0047-random-assignment-of-availability-chunks.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+## Stakeholders
+
+Relay chain node core developers.
+
+## Explanation
+
+### Systematic erasure codes
+
+An erasure coding algorithm is considered systematic if it preserves the original unencoded data as part of the resulting code.
+[The implementation of the erasure coding algorithm used for polkadot's availability data](https://github.com/paritytech/reed-solomon-novelpoly) is systematic. Roughly speaking, the first N_VALIDATORS/3
```
Comment (sandreim) [inline text/0047-random-assignment-of-availability-chunks.md]: That repo is one implementation detail, while this RFC shouldn't be focused on this, rather we should mention reed solomon and the systematic property.

File: text/0047-random-assignment-of-availability-chunks.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+chunks of data can be cheaply concatenated to retrieve the original data, without running the resource-intensive and time-consuming decoding algorithm.
```
Comment (sandreim) [inline text/0047-random-assignment-of-availability-chunks.md]:
```suggestion
chunks of data can be cheaply concatenated to retrieve the original data, without running the resource-intensive and time-consuming reconstruction algorithm.
```

File: text/0047-random-assignment-of-availability-chunks.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+### Availability recovery from systematic chunks
+
+As part of the effort of [increasing polkadot's resource efficiency, scalability and performance](https://github.com/paritytech/roadmap/issues/26), work is under way to
+modify the Availability Recovery subsystem by leveraging systematic chunks. See [this comment](https://github.com/paritytech/polkadot-sdk/issues/598#issuecomment-1792007099)
```
Comment (sandreim) [inline text/0047-random-assignment-of-availability-chunks.md]: I think it is better to describe how is this different than what we have right now.  I think the specific implementation details and tickets should be mentioned in prior art section.

File: text/0047-random-assignment-of-availability-chunks.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+for preliminary performance results.
+
+In this scheme, the relay chain node will first attempt to retrieve the N/3 systematic chunks from the validators that should hold them,
+before falling back to recovering from regular chunks, as before.
+
+The problem that this RFC aims to address is that, currently, the ValidatorIndex is always identical to the ChunkIndex.
+Since the validator array is only shuffled once per session, naively using the ValidatorIndex as the ChunkIndex
+would pose an unreasonable stress on the first N/3 validators during an entire session.
```
Comment (sandreim) [inline text/0047-random-assignment-of-availability-chunks.md]: This looks like a dive deep on the motivation section. Maybe it's worth to compress and move there.

File: text/0047-random-assignment-of-availability-chunks.md (hunk 2 above)
Comment (alindima) [reply text/0047-random-assignment-of-availability-chunks.md]: it can be a little lower. if n_validators is not a power of two, it's actually uses next lower power of two.
